jar-infer-lib/src/main/java/com/uber/nullaway/jarinfer/DefinitelyDerefedParamsDriver.java

```diff
@@ -176,18 +187,30 @@ private static String extractJAR(String jarPath) {
   }
 
   /*
-   * Repack JAR archive and return path
+   * Unpack AAR archive and return 'clasees' directory path
    *
    */
-  private static String packJAR(String jarDir) {
+  private static String extractAAR(String aarPath) {
+    return extractJAR(extractJAR(aarPath) + File.separator + "classes.jar");
```

Comment:
Unpacking and repacking the full `aar` is actually quite expensive, apparently, so in this case at least we should be extracting only `classes.jar` and adding it back to the existing `aar` in memory. See @kageiit 's comment for more details.